src/components/deal-proposal-form/DealProposalForm.tsx

```diff
@@ -164,45 +169,48 @@ export function DealProposalForm({
                 Label
               </HookInput>
 
-              <i className="text-sm font-medium">Current block: {currentBlock}</i>
-              <div className="flex flex-row items-start gap-4">
-                <div className="w-1/2">
-                  <HookInput
-                    id="startBlock"
-                    register={register}
-                    error={errors.startBlock}
-                    type="number"
-                    tooltip="The block number of when the data is guaranteed to be available"
-                  >
-                    Start Block*
-                  </HookInput>
-                </div>
+              <DurationInput
+                name="duration"
+                label="Duration"
+                required={true}
+                control={control}
+                maxMonths={24} // Limit to 2 years
+                error={errors.duration?.months || errors.duration?.days}
+              />
 
-                <DisabledInputInfo
-                  name="startBlockTime"
-                  value={startBlockRealTime.toLocaleString("en-GB")}
-                  label="Estimated real-time"
-                />
-              </div>
+              <Collapsible title="Details">
+                <i className="text-sm font-medium">Current block: {currentBlock}</i>
+                <div className="flex flex-row items-start gap-4">
+                  <div className="w-1/2">
+                    <DisabledInputInfo
+                      name="startBlock"
+                      value={startBlock.toString()}
+                      label="Start Block"
+                    />
+                  </div>
+
+                  <DisabledInputInfo
+                    name="startBlockTime"
+                    value={startBlockRealTime.toLocaleString("en-GB")}
+                    label="Estimated real-time"
+                  />
+                </div>
 
-              <div className="flex flex-row items-start gap-4">
-                <div className="w-1/2">
-                  <HookInput
-                    id="endBlock"
-                    register={register}
-                    error={errors.endBlock}
-                    type="number"
-                    tooltip="The block number when the data ends to be available"
-                  >
-                    End Block*
-                  </HookInput>
+                <div className="flex flex-row items-start gap-4">
+                  <div className="w-1/2">
+                    <DisabledInputInfo
+                      name="endBlock"
+                      value={endBlock.toString()}
+                      label="End Block"
+                    />
+                  </div>
+                  <DisabledInputInfo
+                    name="endBlockTime"
+                    value={endBlockRealTime.toLocaleString("en-GB")}
+                    label="Estimated real-time"
+                  />
                 </div>
```

Comment:
Should be wrapped with `<div className="flex flex-col gap-4">` to get the same gap as the other details